src/dft.cpp

```diff
@@ -494,33 +497,35 @@ double *dft_energy::total() {
   double *Fe = electric();
   double *Fm = magnetic();
   double *F = new double[Nfreq];
-  for (int i = 0; i < Nfreq; ++i) F[i] = Fe[i]+Fm[i];
+  for (int i = 0; i < Nfreq; ++i)
+    F[i] = Fe[i]+Fm[i];
   delete[] Fe;
   delete[] Fm;
   return F;
 }
 
 dft_energy fields::add_dft_energy(const volume_list *where_,
                                   double freq_min, double freq_max, int Nfreq) {
+  
+  if (!where_) // handle empty list of volumes
+    return dft_energy(NULL, NULL, NULL, NULL, freq_min, freq_max, Nfreq, v);
+
   dft_chunk *E = 0, *D = 0, *H = 0, *B = 0;
-  volume_list *where = S.reduce(where_);
+  volume firstvol(where_->v);
+  volume_list *where = new volume_list(where_);
   volume_list *where_save = where;
   while (where) {
     LOOP_OVER_FIELD_DIRECTIONS(gv.dim, d) {
-      E = add_dft(direction_component(Ex, d), where->v, freq_min, freq_max, Nfreq,
-		  true, 1.0, E);
-      D = add_dft(direction_component(Dx, d), where->v, freq_min, freq_max, Nfreq,
-		  true, 1.0, D);
-      H = add_dft(direction_component(Hx, d), where->v, freq_min, freq_max, Nfreq,
-		  true, 1.0, H);
-      B = add_dft(direction_component(Bx, d), where->v, freq_min, freq_max, Nfreq,
-		  true, 1.0, B);
+      E = add_dft(direction_component(Ex, d), where->v, freq_min, freq_max, Nfreq, true, 1.0, E);
+      D = add_dft(direction_component(Dx, d), where->v, freq_min, freq_max, Nfreq, false, 1.0, D);
+      H = add_dft(direction_component(Hx, d), where->v, freq_min, freq_max, Nfreq, true, 1.0, H);
+      B = add_dft(direction_component(Bx, d), where->v, freq_min, freq_max, Nfreq, false, 1.0, B);
```

Comment:
The change to pass `false` for two of the fields is what is fixing your dV factor.   Adding the `where` field to `dft_energy` is unrelated (and is actually unused in this PR).